### PR TITLE
Update story panel ui

### PR DESCRIPTION
**Pull request by zoran995**
### What this PR does

Fixes #<insert issue number here if relevant>

Fix alignment of elements in story panel and move some styling from scss to styled components. This is basically fix of small alignment issues.

Click on the stories button opens a story builder (button on the left from story number)

image
  
### Test me
  Compare
- main - Run story using this [link ](http://ci.terria.io/main/#start=%7B%22version%22%3A%228.0.0%22%2C%22initSources%22%3A%5B%7B%22stratum%22%3A%22user%22%2C%22models%22%3A%7B%7D%2C%22workbench%22%3A%5B%5D%2C%22timeline%22%3A%5B%5D%2C%22initialCamera%22%3A%7B%22west%22%3A100.76792293643028%2C%22south%22%3A-33.86167221726799%2C%22east%22%3A166.232077034218%2C%22north%22%3A-12.73847840362129%2C%22position%22%3A%7B%22x%22%3A-7430867.337576044%2C%22y%22%3A7830500.313937096%2C%22z%22%3A-5368619.004415568%7D%2C%22direction%22%3A%7B%22x%22%3A0.6155177699530678%2C%22y%22%3A-0.6486203927338021%2C%22z%22%3A0.44769349001504455%7D%2C%22up%22%3A%7B%22x%22%3A-0.3081718623601616%2C%22y%22%3A0.32474538372595907%2C%22z%22%3A0.8941870827730343%7D%7D%2C%22homeCamera%22%3A%7B%22west%22%3A109%2C%22south%22%3A-45%2C%22east%22%3A158%2C%22north%22%3A-8%7D%2C%22baseMaps%22%3A%7B%22defaultBaseMapId%22%3A%22basemap-positron%22%7D%2C%22viewerMode%22%3A%223dSmooth%22%2C%22currentTime%22%3A%7B%22dayNumber%22%3A2459656%2C%22secondsOfDay%22%3A12986.616%7D%2C%22showSplitter%22%3Afalse%2C%22splitPosition%22%3A0.5%2C%22stories%22%3A%5B%7B%22title%22%3A%22Rates+of+change+points%22%2C%22text%22%3A%22%3Cdiv%3EDEA+Coastline%27s+rates+of+change+points+estimate+the+rate+at+which+the+coast+is+growing+or+retreating+over+time+at+every+30+metres+along+Australia%27a+entire+non-rocky+coastline.+Labels+next+to+the+points+give+the+average+rate+of+change+in+metres+per+yea+%28negative+values+indicating+retreat+or+erosion%29%2C+with+an+estimate+of+uncertainty+in+brackets.%3C%2Fdiv%3E%3Cdiv%3EWe+can+now+zoom+in+further+to+view+individual+annual+coastlines+since+1988.%3C%2Fdiv%3E%22%2C%22id%22%3A%227e5acea7-2d88-4542-bde0-7810496e2626%22%2C%22shareData%22%3A%7B%22version%22%3A%228.0.0%22%2C%22initSources%22%3A%5B%7B%22stratum%22%3A%22user%22%2C%22models%22%3A%7B%7D%2C%22workbench%22%3A%5B%5D%2C%22timeline%22%3A%5B%5D%2C%22initialCamera%22%3A%7B%22west%22%3A98.10399510124083%2C%22south%22%3A-41.743616235970066%2C%22east%22%3A168.89600486940748%2C%22north%22%3A-4.7344419249184915%2C%22position%22%3A%7B%22x%22%3A-7430867.337576044%2C%22y%22%3A7830500.313937096%2C%22z%22%3A-5368619.004415568%7D%2C%22direction%22%3A%7B%22x%22%3A0.6155177699530678%2C%22y%22%3A-0.6486203927338021%2C%22z%22%3A0.44769349001504455%7D%2C%22up%22%3A%7B%22x%22%3A-0.308171862360162%2C%22y%22%3A0.32474538372595874%2C%22z%22%3A0.8941870827730343%7D%7D%2C%22homeCamera%22%3A%7B%22west%22%3A109%2C%22south%22%3A-45%2C%22east%22%3A158%2C%22north%22%3A-8%7D%2C%22baseMaps%22%3A%7B%22defaultBaseMapId%22%3A%22basemap-positron%22%7D%2C%22viewerMode%22%3A%223dSmooth%22%2C%22currentTime%22%3A%7B%22dayNumber%22%3A2459655%2C%22secondsOfDay%22%3A39601.761%7D%2C%22showSplitter%22%3Afalse%2C%22splitPosition%22%3A0.4999%7D%5D%7D%7D%2C%7B%22title%22%3A%22asd%22%2C%22text%22%3A%22%22%2C%22id%22%3A%22163a033f-a56a-4d02-a2b1-b7c285097dc9%22%2C%22shareData%22%3A%7B%22version%22%3A%228.0.0%22%2C%22initSources%22%3A%5B%7B%22stratum%22%3A%22user%22%2C%22models%22%3A%7B%7D%2C%22workbench%22%3A%5B%5D%2C%22timeline%22%3A%5B%5D%2C%22initialCamera%22%3A%7B%22west%22%3A98.10399510124083%2C%22south%22%3A-41.743616235970066%2C%22east%22%3A168.89600486940748%2C%22north%22%3A-4.7344419249184915%2C%22position%22%3A%7B%22x%22%3A-7430867.337576044%2C%22y%22%3A7830500.313937096%2C%22z%22%3A-5368619.004415568%7D%2C%22direction%22%3A%7B%22x%22%3A0.6155177699530678%2C%22y%22%3A-0.6486203927338021%2C%22z%22%3A0.44769349001504455%7D%2C%22up%22%3A%7B%22x%22%3A-0.308171862360162%2C%22y%22%3A0.32474538372595874%2C%22z%22%3A0.8941870827730343%7D%7D%2C%22homeCamera%22%3A%7B%22west%22%3A109%2C%22south%22%3A-45%2C%22east%22%3A158%2C%22north%22%3A-8%7D%2C%22baseMaps%22%3A%7B%22defaultBaseMapId%22%3A%22basemap-positron%22%7D%2C%22viewerMode%22%3A%223dSmooth%22%2C%22currentTime%22%3A%7B%22dayNumber%22%3A2459655%2C%22secondsOfDay%22%3A39601.761%7D%2C%22showSplitter%22%3Afalse%2C%22splitPosition%22%3A0.4999%7D%5D%7D%7D%5D%7D%5D%7D)
- this branch - run story using this [link ](http://127.0.0.1:3001/#start=%7B%22version%22%3A%228.0.0%22%2C%22initSources%22%3A%5B%7B%22stratum%22%3A%22user%22%2C%22models%22%3A%7B%7D%2C%22workbench%22%3A%5B%5D%2C%22timeline%22%3A%5B%5D%2C%22initialCamera%22%3A%7B%22west%22%3A100.76792293643028%2C%22south%22%3A-33.86167221726799%2C%22east%22%3A166.232077034218%2C%22north%22%3A-12.73847840362129%2C%22position%22%3A%7B%22x%22%3A-7430867.337576044%2C%22y%22%3A7830500.313937096%2C%22z%22%3A-5368619.004415568%7D%2C%22direction%22%3A%7B%22x%22%3A0.6155177699530678%2C%22y%22%3A-0.6486203927338021%2C%22z%22%3A0.44769349001504455%7D%2C%22up%22%3A%7B%22x%22%3A-0.3081718623601616%2C%22y%22%3A0.32474538372595907%2C%22z%22%3A0.8941870827730343%7D%7D%2C%22homeCamera%22%3A%7B%22west%22%3A109%2C%22south%22%3A-45%2C%22east%22%3A158%2C%22north%22%3A-8%7D%2C%22baseMaps%22%3A%7B%22defaultBaseMapId%22%3A%22basemap-positron%22%7D%2C%22viewerMode%22%3A%223dSmooth%22%2C%22currentTime%22%3A%7B%22dayNumber%22%3A2459656%2C%22secondsOfDay%22%3A12986.616%7D%2C%22showSplitter%22%3Afalse%2C%22splitPosition%22%3A0.5%2C%22stories%22%3A%5B%7B%22title%22%3A%22Rates+of+change+points%22%2C%22text%22%3A%22%3Cdiv%3EDEA+Coastline%27s+rates+of+change+points+estimate+the+rate+at+which+the+coast+is+growing+or+retreating+over+time+at+every+30+metres+along+Australia%27a+entire+non-rocky+coastline.+Labels+next+to+the+points+give+the+average+rate+of+change+in+metres+per+yea+%28negative+values+indicating+retreat+or+erosion%29%2C+with+an+estimate+of+uncertainty+in+brackets.%3C%2Fdiv%3E%3Cdiv%3EWe+can+now+zoom+in+further+to+view+individual+annual+coastlines+since+1988.%3C%2Fdiv%3E%22%2C%22id%22%3A%227e5acea7-2d88-4542-bde0-7810496e2626%22%2C%22shareData%22%3A%7B%22version%22%3A%228.0.0%22%2C%22initSources%22%3A%5B%7B%22stratum%22%3A%22user%22%2C%22models%22%3A%7B%7D%2C%22workbench%22%3A%5B%5D%2C%22timeline%22%3A%5B%5D%2C%22initialCamera%22%3A%7B%22west%22%3A98.10399510124083%2C%22south%22%3A-41.743616235970066%2C%22east%22%3A168.89600486940748%2C%22north%22%3A-4.7344419249184915%2C%22position%22%3A%7B%22x%22%3A-7430867.337576044%2C%22y%22%3A7830500.313937096%2C%22z%22%3A-5368619.004415568%7D%2C%22direction%22%3A%7B%22x%22%3A0.6155177699530678%2C%22y%22%3A-0.6486203927338021%2C%22z%22%3A0.44769349001504455%7D%2C%22up%22%3A%7B%22x%22%3A-0.308171862360162%2C%22y%22%3A0.32474538372595874%2C%22z%22%3A0.8941870827730343%7D%7D%2C%22homeCamera%22%3A%7B%22west%22%3A109%2C%22south%22%3A-45%2C%22east%22%3A158%2C%22north%22%3A-8%7D%2C%22baseMaps%22%3A%7B%22defaultBaseMapId%22%3A%22basemap-positron%22%7D%2C%22viewerMode%22%3A%223dSmooth%22%2C%22currentTime%22%3A%7B%22dayNumber%22%3A2459655%2C%22secondsOfDay%22%3A39601.761%7D%2C%22showSplitter%22%3Afalse%2C%22splitPosition%22%3A0.4999%7D%5D%7D%7D%2C%7B%22title%22%3A%22asd%22%2C%22text%22%3A%22%22%2C%22id%22%3A%22163a033f-a56a-4d02-a2b1-b7c285097dc9%22%2C%22shareData%22%3A%7B%22version%22%3A%228.0.0%22%2C%22initSources%22%3A%5B%7B%22stratum%22%3A%22user%22%2C%22models%22%3A%7B%7D%2C%22workbench%22%3A%5B%5D%2C%22timeline%22%3A%5B%5D%2C%22initialCamera%22%3A%7B%22west%22%3A98.10399510124083%2C%22south%22%3A-41.743616235970066%2C%22east%22%3A168.89600486940748%2C%22north%22%3A-4.7344419249184915%2C%22position%22%3A%7B%22x%22%3A-7430867.337576044%2C%22y%22%3A7830500.313937096%2C%22z%22%3A-5368619.004415568%7D%2C%22direction%22%3A%7B%22x%22%3A0.6155177699530678%2C%22y%22%3A-0.6486203927338021%2C%22z%22%3A0.44769349001504455%7D%2C%22up%22%3A%7B%22x%22%3A-0.308171862360162%2C%22y%22%3A0.32474538372595874%2C%22z%22%3A0.8941870827730343%7D%7D%2C%22homeCamera%22%3A%7B%22west%22%3A109%2C%22south%22%3A-45%2C%22east%22%3A158%2C%22north%22%3A-8%7D%2C%22baseMaps%22%3A%7B%22defaultBaseMapId%22%3A%22basemap-positron%22%7D%2C%22viewerMode%22%3A%223dSmooth%22%2C%22currentTime%22%3A%7B%22dayNumber%22%3A2459655%2C%22secondsOfDay%22%3A39601.761%7D%2C%22showSplitter%22%3Afalse%2C%22splitPosition%22%3A0.4999%7D%5D%7D%7D%5D%7D%5D%7D) - after running this branch locally 
It seems that putting story into init file on gist and referencing is not loading it

### Checklist

-   [ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [ ] I've updated relevant documentation in `doc/`.
-   [ ] I've updated CHANGES.md with what I changed.
-   [ ] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

File: lib/ReactViews/Story/StoryPanel/StoryButtons.tsx [hunk 1]
```diff
@@ -13,39 +15,38 @@ export const CollapseBtn = ({
...
-    <button
-      className={Styles.exitBtn}
+    <RawButton
       title={isCollapsed ? t("story.expand") : t("story.collapse")}
       onClick={onClick}
     >
-      <Icon glyph={isCollapsed ? Icon.GLYPHS.info : Icon.GLYPHS.arrowDown} />
-    </button>
+      {isCollapsed ? (
+        <StoryIcon styledWidth={"24px"} glyph={Icon.GLYPHS.info} />
```
Comment (nf-s) [inline lib/ReactViews/Story/StoryPanel/StoryButtons.tsx]: If we are trying to preserve the original story panel design, the icon sizes are slightly different.

### Before after

image


```suggestion
        <StoryIcon styledWidth={"15px"} glyph={Icon.GLYPHS.info} />
```

File: lib/ReactViews/Story/StoryPanel/StoryButtons.tsx [hunk 2]
```diff
@@ -13,39 +15,38 @@ export const CollapseBtn = ({
...
+      ) : (
+        <StoryIcon styledWidth={"12px"} glyph={Icon.GLYPHS.arrowDown} />
```
Comment (nf-s) [inline lib/ReactViews/Story/StoryPanel/StoryButtons.tsx:25]:
```suggestion
        <StoryIcon styledWidth={"15px"} glyph={Icon.GLYPHS.arrowDown} />
```

File: lib/ReactViews/Story/StoryPanel/StoryButtons.tsx [hunk 3]
```diff
@@ -13,39 +15,38 @@ export const CollapseBtn = ({
...
-      onClick={onClick}
-    >
-      <Icon glyph={Icon.GLYPHS.location} />
-    </button>
-  );
-};
+export const StoryIcon = styled(StyledIcon).attrs(props => ({
+  fillColor: props.theme.textDark,
+  opacity: 0.5
+}))``;
```
Comment (nf-s) [inline lib/ReactViews/Story/StoryPanel/StoryButtons.tsx]: Is there an easy way to add `hover` styles here?
Something like

```suggestion
export const StoryIcon = styled(StyledIcon).attrs(props => ({
  fillColor: props.theme.textDark,
  opacity: 0.5
}))`
  &:hover {
    fill: ${p => p.theme.colorPrimary};
  }
`;
```

File: lib/ReactViews/Story/StoryPanel/StoryButtons.tsx [hunk 4]
```diff
@@ -13,39 +15,38 @@ export const CollapseBtn = ({
...
-    </button>
+    <RawButton onClick={onClick} title={t("story.exitBtn")}>
+      <StoryIcon
+        styledWidth={"12px"}
+        glyph={Icon.GLYPHS.close}
+        css={`
+          border-radius: 50%;
+          border: 3px solid ${theme.textDark};
+          padding: 3px;
+        `}
```
Comment (nf-s) [inline lib/ReactViews/Story/StoryPanel/StoryButtons.tsx]: I can't remember original story panel design - do we want to keep the circle border around "X" close button here @philipgrimmett ?

### Before vs After

image

**Review by nf-s — APPROVED** [review]
Thanks @zoran995 for the PR!

I have added some comments, let me know what you think - but I am happy with the PR in it's current state

File: lib/ReactViews/Story/StoryPanel/StoryButtons.tsx (hunk 4 above)
Comment (zoran995) [reply lib/ReactViews/Story/StoryPanel/StoryButtons.tsx]: I took it from https://github.com/TerriaJS/terriajs/issues/5427#issuecomment-1032151750
https://www.figma.com/proto/0IrsjWUaA3QPYPNtDcoiNr/Terria---Stories---Compromise----Feb22?page-id=0%3A1&node-id=1%3A421&viewport=241%2C48%2C0.66&scaling=min-zoom&starting-point-node-id=1%3A421

File: lib/ReactViews/Story/StoryPanel/StoryButtons.tsx (hunk 1 above)
Comment (zoran995) [reply lib/ReactViews/Story/StoryPanel/StoryButtons.tsx]: changed it to 20px as per @philipgrimmett design, let me know if it needs to be updated (missed the inspect link before)

File: lib/ReactViews/Story/StoryPanel/StoryButtons.tsx (hunk 2 above)
Comment (zoran995) [reply lib/ReactViews/Story/StoryPanel/StoryButtons.tsx:25]: per phils design it should be 10px but I found it to small so put it to 12px, feel free to directly commit the change 
15px
image
12px
image

File: lib/ReactViews/Story/StoryPanel/StoryButtons.tsx (hunk 3 above)
Comment (zoran995) [reply lib/ReactViews/Story/StoryPanel/StoryButtons.tsx]: done, also added hover to close button border-color

File: lib/ReactViews/Story/StoryPanel/StoryButtons.tsx (hunk 4 above)
Comment (nf-s) [reply lib/ReactViews/Story/StoryPanel/StoryButtons.tsx]: Ah cool! Sorry I forgot about that prototype.
Looks good!

File: lib/ReactViews/Story/StoryPanel/StoryButtons.tsx (hunk 1 above)
Comment (nf-s) [reply lib/ReactViews/Story/StoryPanel/StoryButtons.tsx]: Looks good!

**Comment by nf-s** [comment]
Awesome, thanks @zoran995 !

Sorry I forgot about that Figma prototype - everything looks good!

**Pull request merged**